app/src/main/java/openfoodfacts/github/scrachx/openfood/views/HistoryScanActivity.java

```diff
@@ -185,9 +192,33 @@ public void exportCSV() {
             }
             writer.close();
             Toast.makeText(this, R.string.txt_history_exported, Toast.LENGTH_LONG).show();
+            isDownload = true;
         } catch (IOException e) {
             e.printStackTrace();
         }
+
+        NotificationManager notificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
+        Intent downloadIntent = new Intent(Intent.ACTION_VIEW);
+        downloadIntent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
+        Uri csvUri = FileProvider.getUriForFile(this, this.getPackageName() + ".provider", f);
+        downloadIntent.setDataAndType(csvUri, "text/csv");
+        downloadIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
+            int importance = NotificationManager.IMPORTANCE_DEFAULT;
+            NotificationChannel notificationChannel = new NotificationChannel("downloadChannel", "ChannelCSV", importance);
```

Comment:
Should the notification channel not be translateable too? To my knowledge, wouldn't this show in the app settings on the users device under notifications?

Reply:
@Karljoones I did not quite see that it would show up in the settings .Thanks for pointing it out . I will make the change .

Reply:
Thanks! All is well once that is fixed!

Reply:
@Karljoones  Notification channels have options for lights and vibrations.
Do you want this notification channel to have lights and vibration . 
What level of importance do you want this channel to be ?

Reply:
I think just the defaults would be fine for this.

Reply:
What channel name should i keep >

Reply:
Maybe "Export notifications" or something along those lines.

Reply:
I have kept the name you suggested . Thanks